### PR TITLE
feat: reusable workflow to allow testing on infra team's cluster through teleport

### PR DESCRIPTION
**Pull request by bkenez**
Implements https://github.com/camunda/team-infrastructure-experience/issues/339

Currently support testing for latest stable and snapshot, but only with the new operational procedure for failover and failback.

### CODE REVIEW

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+---
+name: Nightly Teleport EKS Operational Procedure Test Stable
+
+on:
+    schedule:
+        - cron: 0 2 * * 2 # Every Tuesday at 02:00 UTC
+    push:
+        branches:
+            - reusable-gh-workflow
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:9]: the push trigger can go

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            fail-fast: false
+            matrix:
+                helmChartVersion: ${{fromJson(needs.dynamic-matrix.outputs.matrix).helmChartVersion}}
+        uses: ./.github/workflows/reusable_teleport_operational_procedure.yml
+        with:
+            helmChartVersion: ${{ matrix.helmChartVersion }}
+            helmChartName: oci://ghcr.io/camunda/helm/camunda-platform
+            globalImageTag: SNAPSHOT
+            awsProfile: infex
+            testsTfBinaryName: terraform
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:80]: these are also defined as global env, do we need the global env then still or can we pass them in here instead of double defining?

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                  echo "${versions}"
+                  echo "matrix=${versions}" >> "$GITHUB_OUTPUT"
+
+    operational-procedure:
+        needs:
+            - dynamic-matrix
+        strategy:
+            # parallelism is limited due to infra cluster resource constraints.
+            # Once multiple versions are compatible with testing through teleport, investigate if increasing is possible.
+            max-parallel: 1
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:70]: What are the constraints and why are we limited it by it?

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            clusterName: camunda-ci-eks
+            labels: janitor/ttl=1h camunda.cloud/ephemeral=true
+            cluster: camunda.teleport.sh-camunda-ci-eks
+            backupBucket: tests-c8-multi-region-es-eu-central-1
+            helmReleaseName: camunda
+            zeebeClusterSize: 8
+        secrets:
+            VAULT_ADDR: ${{ secrets.VAULT_ADDR }}
+            VAULT_ROLE_ID: ${{ secrets.VAULT_ROLE_ID }}
+            VAULT_SECRET_ID: ${{ secrets.VAULT_SECRET_ID }}
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:91]: could be simplified by `secrets: inherit`

[reference](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idsecretsinherit)

File: .github/workflows/nightly_teleport_operations_procedure_stable.yml [hunk 5]
```diff
@@ -0,0 +1,93 @@
+---
+name: Nightly Teleport EKS Operational Procedure Test Stable
+
+on:
+    schedule:
+        - cron: 0 2 * * 2 # Every Tuesday at 02:00 UTC
+    push:
+        branches:
+            - reusable-gh-workflow
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operations_procedure_stable.yml:9]: push trigger

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
+---
+name: Nightly Teleport EKS Operational Procedure Test Stable
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:2]: Let's remove the nightly, the UI of GitHub is still a bit meh. Due to that maybe also rearrange the order.
The name is anyway wrong since it's snapshot but called stable here.

I'd suggest with something like:

```
Snapshot Teleport EKS Operational Procedure
```

Or similar, feel free to change it up.

The reasoning behind it:
Based on that one would have to click on everything to get where they wanted.
image

File: .github/workflows/nightly_teleport_operations_procedure_stable.yml [hunk 7]
```diff
@@ -0,0 +1,93 @@
+---
+name: Nightly Teleport EKS Operational Procedure Test Stable
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operations_procedure_stable.yml:2]: Same as the other comment on the workflow to rearrange the naming and remove nightly.
Just for better visualization of the new workflows.

File: .github/workflows/nightly_teleport_operations_procedure_stable.yml [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    dynamic-matrix:
+        runs-on: ubuntu-latest
+        outputs:
+            matrix: ${{ steps.generate-matrix.outputs.matrix }}
+        steps:
+            - name: Generate matrix
+              id: generate-matrix
+              run: |
+                  # renovate: datasource=helm depName=camunda-platform registryUrl=https://helm.camunda.io versioning=regex:^11(\.(?<minor>\d+))?(\.(?<patch>\d+))?$
+                  c86=11.1.1
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operations_procedure_stable.yml:44]: maybe we could bump it to `11.2.0` already to align with the other workflow?
Worst case Renovate will eventually pick it up

File: .github/workflows/nightly_teleport_operations_procedure_stable.yml [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+                  # Snapshot is used for namespace setup to do the DNS chaining tests with a static namespace. It's not used for the matrix.
+                  # shellcheck disable=SC2034
+                  c87=SNAPSHOT
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operations_procedure_stable.yml:48]: does this also apply to the teleport workflows?

File: .github/workflows/nightly_teleport_operations_procedure_stable.yml [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+            clusterName: camunda-ci-eks
+            labels: janitor/ttl=1h camunda.cloud/ephemeral=true
+            cluster: camunda.teleport.sh-camunda-ci-eks
+            backupBucket: tests-c8-multi-region-es-eu-central-1
+            helmReleaseName: camunda
+            zeebeClusterSize: 8
+        secrets:
+            VAULT_ADDR: ${{ secrets.VAULT_ADDR }}
+            VAULT_ROLE_ID: ${{ secrets.VAULT_ROLE_ID }}
+            VAULT_SECRET_ID: ${{ secrets.VAULT_SECRET_ID }}
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operations_procedure_stable.yml:93]: same topic, could be simplified by inherit if wanted

File: .github/workflows/nightly_teleport_operations_procedure_stable.yml [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+            # Once multiple versions are compatible with testing through teleport, investigate if increasing is possible.
+            max-parallel: 1
+            fail-fast: false
+            matrix:
+                helmChartVersion: ${{fromJson(needs.dynamic-matrix.outputs.matrix).helmChartVersion}}
+        uses: ./.github/workflows/reusable_teleport_operational_procedure.yml
+        with:
+            helmChartVersion: ${{ matrix.helmChartVersion }}
+            awsProfile: infex
+            testsTfBinaryName: terraform
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operations_procedure_stable.yml:82]: same question as in the snapshot workflow about global env vs here hardcoded again

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+
+on:
+    schedule:
+        - cron: 0 2 * * 2 # Every Tuesday at 02:00 UTC
```
Comment (Langleu) [inline .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:6]: if resources are such an issue, maybe it makes sense to not run both at the same time on  a weekly basis?
They would run in parallel and if we limit to 1 execution, they would clash on cluster deploy and potentially starve each other if resources are an issue.

File: .github/workflows/reusable_teleport_operational_procedure.yml [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+
+            - name: Setup AWS and Tools
+              uses: ./.github/actions/setup-aws
+              with:
+                  secrets: ${{ toJSON(secrets) }}
+
+            - name: Set up Teleport
+              uses: teleport-actions/setup@v1
+              with:
+                  version: 16.4.11
```
Comment (Langleu) [inline .github/workflows/reusable_teleport_operational_procedure.yml:158]: The version looks quite outdated to me, e.g. Infra is on
https://github.com/camunda/infra-core/blob/stage/.tool-versions#L43

I'm wondering whether it makes sense for Infra to supply a reuseable action just for the `Set up Teleport` step which they maintain.

The problem where I'm coming from that either:
a) we maintain it manually to align with what Infra has, so lagging behind potentially
b) Renovate maintains it after adding the required notions to pick it up but then we may be ahead of Infra. E.g. Major release, Infra takes time to do it, we might be incompatible and would block Renovate PRs till it's merged at Infra.
c) if they maintain the versioning, we could just reference the `main` branch to always use the compatible version that they provide / use.

File: .github/workflows/reusable_teleport_operational_procedure.yml [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+
+            - name: Import Secrets
+              id: secrets
+              uses: hashicorp/vault-action@v3
+              with:
+                  url: ${{ secrets.VAULT_ADDR }}
+                  method: approle
+                  roleId: ${{ secrets.VAULT_ROLE_ID }}
+                  secretId: ${{ secrets.VAULT_SECRET_ID }}
+                  exportEnv: true
```
Comment (Langleu) [inline .github/workflows/reusable_teleport_operational_procedure.yml:200]: if you're not using them as env variables anywhere, I would not export them

File: .github/workflows/reusable_teleport_operational_procedure.yml [hunk 15]
```diff
@@ -0,0 +1,276 @@
...
+                  set -euxo pipefail
+                  printf -v now '%(%s)T'
+                  duration=$((now - ${{ steps.failback-start.outputs.timestamp }}))
+                  echo $duration
+                  if [ "$duration" -gt "1500" ]; then
+                  echo "::error ::Failback of ${{ inputs.helmChartVersion }} is taking longer than 25 minutes"
+                  fi
+
+            - name: Delete namespaces
+              if: ${{ always() }}
```
Comment (Langleu) [inline .github/workflows/reusable_teleport_operational_procedure.yml:271]: just as side note, you can also just do `if: always()`

[reference](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#about-expressions)

File: test/internal/helpers/kubectl/helpers.go [hunk 16]
```diff
@@ -244,9 +235,38 @@ func RunSensitiveKubectlCommand(t *testing.T, kubectlOptions *k8s.KubectlOptions
...
-	output, err := k8s.RunKubectlAndGetOutputE(t, &cluster.KubectlNamespace, "exec", "camunda-elasticsearch-master-0", "--", "curl", "-XPUT", "http://localhost:9200/_snapshot/camunda_backup", "-H", "Content-Type: application/json", "-d", fmt.Sprintf("{\"type\": \"s3\", \"settings\": {\"bucket\": \"%s-elastic-backup\", \"client\": \"camunda\", \"base_path\": \"%s-backups\"}}", clusterName, version))
+	// Determine if Teleport mode is enabled.
+	teleportEnabled := false
+	if teleportStr, ok := os.LookupEnv("TELEPORT"); ok {
+		if parsed, err := strconv.ParseBool(teleportStr); err == nil {
+			teleportEnabled = parsed
+		} else {
+			t.Fatalf("[ELASTICSEARCH] Failed to parse TELEPORT env var: %v", err)
+		}
+	}
```
Comment (Langleu) [inline test/internal/helpers/kubectl/helpers.go:249]: The following is just a suggestion and preference, just a quick idea but feel free to dismiss it.

we could align it with the rest of the code and use the helper function to retrieve the value?

https://github.com/camunda/c8-multi-region/blob/main/test/internal/helpers/helpers.go#L27-L33

```
teleportEnabled := helpers.GetEnv("TELEPORT", "false")
```

simplified everything a bit since you don't have to convert to boolean and since you control the default value, anything defining `TELEPORT` would be true.

```
if teleportEnabled != "false" {
....
}
```

File: test/internal/helpers/kubectl/helpers.go [hunk 17]
```diff
@@ -340,11 +360,24 @@ func createZeebeContactPoints(t *testing.T, size int, namespace0, namespace1 str
...
+
+	// Check if TELEPORT is enabled.
+	teleportEnabled := false
+	if teleportStr := os.Getenv("TELEPORT"); teleportStr != "" {
+		var err error
+		teleportEnabled, err = strconv.ParseBool(teleportStr)
+		if err != nil {
+			t.Fatalf("[ELASTICSEARCH] Failed to parse TELEPORT env var: %v", err)
+		}
+	}
```
Comment (Langleu) [inline test/internal/helpers/kubectl/helpers.go:372]: same as above suggestion on the value

File: test/multi_region_aws_camunda_test.go [hunk 18]
```diff
@@ -173,52 +177,100 @@ func TestAWSDualRegCleanup(t *testing.T) {
...
 
+	// Check if TELEPORT is enabled.
+	teleportEnabled := false
+	if teleportStr := os.Getenv("TELEPORT"); teleportStr != "" {
+		var err error
+		teleportEnabled, err = strconv.ParseBool(teleportStr)
+		if err != nil {
+			t.Fatalf("[ELASTICSEARCH] Failed to parse TELEPORT env var: %v", err)
+		}
+	}
```
Comment (Langleu) [inline test/multi_region_aws_camunda_test.go:234]: same teleport topic.

File: test/multi_region_aws_camunda_test.go [hunk 19]
```diff
@@ -173,52 +177,100 @@ func TestAWSDualRegCleanup(t *testing.T) {
...
-		KubectlSystem:    *k8s.NewKubectlOptions("", kubeConfigPrimary, "kube-system"),
-		KubectlFailover:  *k8s.NewKubectlOptions("", kubeConfigPrimary, primaryNamespaceFailover),
+	// Determine if Teleport mode is enabled.
+	teleportEnabled := false
+	if teleportStr, ok := os.LookupEnv("TELEPORT"); ok {
+		if parsed, err := strconv.ParseBool(teleportStr); err == nil {
+			teleportEnabled = parsed
+		} else {
+			t.Fatalf("failed to parse TELEPORT env var: %v", err)
+		}
```
[the remark this answers is not in the document]
Comment (Langleu) [reply test/multi_region_aws_camunda_test.go:187]: you could even define `teleportEnabled` as var on package level and don't have to redo everything every time.
You could just simply go straight to your ifs

File: test/multi_region_aws_camunda_test.go [hunk 20]
```diff
@@ -173,52 +177,100 @@ func TestAWSDualRegCleanup(t *testing.T) {
...
+	retries := 30
+	var setValues map[string]string
+
+	if teleportEnabled {
+		timeout = "1800s"
+		retries = 100
+		setValues = map[string]string{
+			"zeebe.affinity": "null",
+		}
+	}
```
Comment (Langleu) [inline test/multi_region_aws_camunda_test.go:246]: why overcomplicate the setValues topic.

Pseudocode
```
if teleportEnabled {
timeout ...
retries..

baseHelmVars["zeebe.affinity"] = "null"
`
```

`baseHelmVars` is a global package var, so you can just add it to it. Even if you do it later again, no problem in overwriting or setting it again if you really want.
`baseHelmVars` is always passed as parameter

File: test/multi_region_aws_camunda_test.go [hunk 21]
```diff
@@ -308,14 +360,36 @@ func deleteSecondaryRegion(t *testing.T) {
...
+			"zeebe.affinity":   "null",
+			"operate.enabled":  "false",
+			"tasklist.enabled": "false",
+		}
+	} else {
+		setValues = map[string]string{
+			"operate.enabled":  "false",
+			"tasklist.enabled": "false",
+		}
 	}
```
Comment (Langleu) [inline test/multi_region_aws_camunda_test.go:388]: same affinity topic as above.
You wouldn't even need it if you had added it in deploy already.

File: test/multi_region_aws_camunda_test.go [hunk 22]
```diff
@@ -308,14 +360,36 @@ func deleteSecondaryRegion(t *testing.T) {
...
+	teleportEnabled := false
+	if teleportStr := os.Getenv("TELEPORT"); teleportStr != "" {
+		var err error
+		teleportEnabled, err = strconv.ParseBool(teleportStr)
+		if err != nil {
+			t.Fatalf("[ELASTICSEARCH] Failed to parse TELEPORT env var: %v", err)
+		}
+	}
+
+	timeout := "600s"
```
Comment (Langleu) [inline test/multi_region_aws_camunda_test.go:373]: seeing as 2nd occurence, maybe makes sense to de-duplicate it and pull out to package level var

File: test/multi_region_aws_camunda_test.go [hunk 23]
```diff
@@ -395,7 +469,26 @@ func scaleUpWebApps(t *testing.T) {
...
+		}
+	}
+
+	var setValues map[string]string
+
+	if teleportEnabled {
+		setValues = map[string]string{
+			"zeebe.affinity": "null",
+		}
+	}
```
Comment (Langleu) [inline test/multi_region_aws_camunda_test.go:489]: becomes obsolete if handled in the `baseHelmVars`

**Review by Langleu — CHANGES_REQUESTED** [review]
Left some comments

e.g. `teleportEnabled` as mentioned could be pulled out to the package level, e.g. keep it with everything else in `multi_region_aws_camunda_test.go`, so you wouldn't have to retrieve it every time.
Some other optimization proposals.

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml (hunk 1 above)
Comment (bkenez) [reply .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:9]: Right, that was just for testing, removing.

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml (hunk 2 above)
Comment (bkenez) [reply .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:80]: That was a leftover - I intended for all variables to be defined in the actual workflow call. Removed.

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml (hunk 3 above)
Comment (bkenez) [reply .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:70]: It is somewhat strange, but even with the extra timeouts put in place and anti-affinity removed, when I started two or more in parallel, they were a lot more likely to time out. The Infra team said they intend to increase the capacity, so I figure, for now, stip parallelism to guarantee test execution.

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml (hunk 4 above)
Comment (bkenez) [reply .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:91]: True, fixed. 👍

File: .github/workflows/nightly_teleport_operations_procedure_stable.yml (hunk 5 above)
Comment (bkenez) [reply .github/workflows/nightly_teleport_operations_procedure_stable.yml:9]: Fixed alongside the above one

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml (hunk 6 above)
Comment (bkenez) [reply .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:2]: Good point, I also thought that was annoying

File: .github/workflows/nightly_teleport_operations_procedure_stable.yml (hunk 8 above)
Comment (bkenez) [reply .github/workflows/nightly_teleport_operations_procedure_stable.yml:44]: True, we got a new version while I was working on it 😄

File: .github/workflows/nightly_aws_operational_procedure_stable.yml [hunk 24]
```diff
@@ -151,8 +151,6 @@ jobs:
             - name: Generate matrix
               id: generate-matrix
               run: |
-                  # renovate: datasource=helm depName=camunda-platform registryUrl=https://helm.camunda.io versioning=regex:^9(\.(?<minor>\d+))?(\.(?<patch>\d+))?$
-                  c84=9.4.7
                   # renovate: datasource=helm depName=camunda-platform registryUrl=https://helm.camunda.io versioning=regex:^10(\.(?<minor>\d+))?(\.(?<patch>\d+))?$
                   c85=10.4.8
                   # renovate: datasource=helm depName=camunda-platform registryUrl=https://helm.camunda.io versioning=regex:^11(\.(?<minor>\d+))?(\.(?<patch>\d+))?$
```
[the remark this answers is not in the document]
Comment (bkenez) [reply .github/workflows/nightly_aws_operational_procedure_stable.yml:156]: Good call about not testing new procedure 8.6 twice. 
I thought to leave in the test and k8s triggers just in case it would somehow have an edge case that would fail the terraform version, but we would learn about that anywhen when it runs - not much benefit confirming after every single change,  true.

File: .github/workflows/nightly_teleport_operational_procedure_snapshot.yml (hunk 12 above)
Comment (bkenez) [reply .github/workflows/nightly_teleport_operational_procedure_snapshot.yml:6]: Right, I copied one as a template but hadn't updated the day.

File: .github/workflows/reusable_teleport_operational_procedure.yml [hunk 25]
```diff
@@ -0,0 +1,276 @@
...
+name: reusable_teleport_operational_procedure
+
+on:
+    workflow_call:
+        inputs:
+            helmChartVersion:
+                type: string
+                description: Version of the Helm chart to deploy.
+                required: false
+                default: 11.1.0
```
[the remark this answers is not in the document]
Comment (bkenez) [reply .github/workflows/reusable_teleport_operational_procedure.yml:11]: Fixed 👍

File: .github/workflows/reusable_teleport_operational_procedure.yml (hunk 13 above)
Comment (bkenez) [reply .github/workflows/reusable_teleport_operational_procedure.yml:158]: This was also a point of doubt for me: I will reach out to infra to see if they have any recommendations regarding this point - I would say the reusable action from their side would be the best.

File: .github/workflows/reusable_teleport_operational_procedure.yml (hunk 14 above)
Comment (bkenez) [reply .github/workflows/reusable_teleport_operational_procedure.yml:200]: Right, removed. Now I remember you mentioned this before too, but I forgot to, then 😅

File: test/multi_region_aws_camunda_test.go (hunk 20 above)
Comment (bkenez) [reply test/multi_region_aws_camunda_test.go]: That was caused by me fixing recreateCamundaInSecondary_8_6_plus first and copying the setValues way of doing things, assuming that is the appropriate way 😅  But thanks for this; I really managed to simplify it now.

File: test/multi_region_aws_camunda_test.go (hunk 21 above)
Comment (bkenez) [reply test/multi_region_aws_camunda_test.go]: I simplified it to modify affinity through baseValues; however, should I also replace the original setValues way of doing things with it while I'm there?

**Comment by bkenez** [comment]
Thanks for the review! Appreciate the corrections and advice ❤️
I resolved the teleportEnabled issue by moving the conversion to a helper function. This keeps the logic cleaner by directly converting the env var to a boolean, avoiding unnecessary variables.
Coming from Python I needed some time to get used to strong typing again. Hope this approach works better. Let me know if you see any improvements still.

File: .github/workflows/reusable_teleport_operational_procedure.yml (hunk 13 above)
Comment (Langleu) [reply .github/workflows/reusable_teleport_operational_procedure.yml:153]: if not, we should tag it appropriately with the renovate stuff so it's automatically updated except for major versions (Default).

File: .github/workflows/nightly_aws_operational_procedure_stable.yml [hunk 26]
```diff
@@ -167,7 +161,7 @@ jobs:
                   c87=SNAPSHOT
 
                   if [ "${{ inputs.helm-versions }}" == "" ]; then
-                    versions='{"helm-version":["'${c84}'","'${c85}'","'${c86}'","'OLD.${c86_old}'"]}'
+                    versions='{"helm-version":["'${c85}'","'${c86}'","'OLD.${c86_old}'"]}'
```
Comment (Langleu) [inline .github/workflows/nightly_aws_operational_procedure_stable.yml:164]: c86 is undefined now

File: test/internal/helpers/kubectl/helpers.go [hunk 27]
```diff
@@ -340,11 +350,14 @@ func createZeebeContactPoints(t *testing.T, size int, namespace0, namespace1 str
...
-	os.Setenv("HELM_RELEASE_NAME", "camunda")
-	os.Setenv("ZEEBE_CLUSTER_SIZE", "8")
+
+	if helpers.IsTeleportEnabled() {
+		// Set environment variables for the script
+		os.Setenv("CAMUNDA_NAMESPACE_0", namespace0)
+		os.Setenv("CAMUNDA_NAMESPACE_1", namespace1)
+		os.Setenv("HELM_RELEASE_NAME", "camunda")
+		os.Setenv("ZEEBE_CLUSTER_SIZE", "8")
+	}
```
Comment (Langleu) [inline test/internal/helpers/kubectl/helpers.go:360]: correct me if I'm wrong but wasn't this a negation before?

**Review by Langleu — CHANGES_REQUESTED** [review]
Looks a lot cleaner 👍 , thanks for the changes.
Just 2 smaller things + the teleport version topic.

File: test/multi_region_aws_camunda_test.go (hunk 21 above)
Comment (Langleu) [reply test/multi_region_aws_camunda_test.go]: Not sure I can follow about what you want to change?

Comment (bkenez) [reply test/multi_region_aws_camunda_test.go]: Talked it over in our sync meeting: the remainder is fine as-is

File: .github/workflows/reusable_teleport_operational_procedure.yml (hunk 13 above)
Comment (bkenez) [reply .github/workflows/reusable_teleport_operational_procedure.yml:154]: Implemented suggestions from the infra team, including their custom datasource. This is their final solution for now, but there might be further developments down the line.
